pkgs/dkg/drand.go

```diff
@@ -214,77 +360,199 @@ func (o *LocalOwner) PostDKG(res *dkg.OptionResult) error {
 	o.Logger.Info("DKG ceremony finished successfully")
 	// Store result share a instance
 	o.SecretShare = res.Result.Key
-
+	// encode priv share
+	secret := &DistKeyShare{}
+	var commits []byte
+	for _, point := range o.SecretShare.Commits {
+		b, _ := point.MarshalBinary()
+		commits = append(commits, b...)
+	}
+	secret.Commits = commits
+	secterPoint, err := o.SecretShare.Share.V.MarshalBinary()
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	secret.Share.V = secterPoint
+	secret.Share.I = o.SecretShare.Share.I
+	bin, err := secret.Encode()
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	encBin, err := o.EncryptSecretDB(bin)
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	err = o.DB.Set([]byte("secret"), o.Data.ReqID[:], encBin)
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
 	// Get validator BLS public key from result
-	validatorPubKey, err := crypto.ResultToValidatorPK(res.Result, o.suite.G1().(dkg.Suite))
+	validatorPubKey, err := crypto.ResultToValidatorPK(res.Result, o.Suite.G1().(dkg.Suite))
 	if err != nil {
 		o.broadcastError(err)
 		return err
 	}
 	o.Logger.Debug("Validator`s public key %x", zap.String("key", fmt.Sprintf("%x", validatorPubKey.Serialize())))
-
 	// Get BLS partial secret key share from DKG
 	secretKeyBLS, err := crypto.ResultToShareSecretKey(res.Result)
 	if err != nil {
 		o.broadcastError(err)
 		return err
 	}
 	// Store secret if requested
-	if viper.GetBool("storeShare") {
-		type shareStorage struct {
-			Index  int    `json:"index"`
-			Secret string `json:"secret"`
-		}
-		data := shareStorage{
-			Index:  res.Result.Key.Share.I,
-			Secret: secretKeyBLS.SerializeToHexStr(),
+	if StoreShare {
+		ciphertext, err := o.encryptSecretShare(secretKeyBLS)
+		if err != nil {
+			o.Logger.Error("Cant encrypt secret share: ", zap.Error(err))
+			o.broadcastError(err)
+			return err
 		}
-		err = utils.WriteJSON("./secret_share_"+hex.EncodeToString(o.data.ReqID[:]), &data)
+		err = utils.StoreSecretShareToFile(OutputPath, res.Result.Key.Share.I, ciphertext, validatorPubKey)
```

Comment:
same as I said about the db, I don't like the fact we're writing files from here. so far we returned from here and used higher level code to store stuff. I want to keep this code as pure protocol and not do any I/O.

Reply:
Moved higher level to operator/state package.